### PR TITLE
Update 'master' to use Cluster Pool

### PR DESCRIPTION
**Pull request by kpouget**
The rest of `ci-artifacts` will follow once we confirm that this works as expected.

### CODE REVIEW

**Comment by kpouget** [comment]
@omertuc @dagrayvid @ArangoGutierrez FYI, this PR (and the following ones) should speed up the testing of `ci-artifacts` by 1/ giving a cluster from a pre-populated pool (great for PR), and 2/ avoid any cluster installation issue (less nightly "infrastructure" flakes)

I test first witht the GPU Operator, then I'll update all the other tests (Steve confirmed that's the recommended way to proceed)

File: ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__4.6.yaml [hunk 1]
```diff
@@ -27,11 +21,16 @@ resources:
       memory: 2Gi
 tests:
 - as: gpu-operator-e2e
+  cluster_claim:
+    architecture: amd64
+    cloud: aws
+    owner: openshift-ci
+    product: ocp
+    timeout: 1h0m0s
```
Comment (omertuc) [inline ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__4.6.yaml:29]: This sounds a little bit too strict, 

image

Comment (kpouget) [reply ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__4.6.yaml]: hum indeed, I've changed it to 1h30
```
INFO[2021-07-22T22:10:45Z] Running step gpu-operator-e2e-presubmit-operatorhub. 
INFO[2021-07-22T22:38:55Z] Step gpu-operator-e2e-presubmit-operatorhub succeeded after 28m10s. 
INFO[2021-07-22T22:38:55Z] Running step gpu-operator-e2e-presubmit-operatorhub-cleanup. 
INFO[2021-07-22T22:39:45Z] Step gpu-operator-e2e-presubmit-operatorhub-cleanup succeeded after 50s. 
INFO[2021-07-22T22:39:45Z] Running step gpu-operator-e2e-presubmit-master. 
INFO[2021-07-22T23:04:26Z] Step gpu-operator-e2e-presubmit-master succeeded after 24m40s. 
```


https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-psap_ci-artifacts/214/pull-ci-openshift-psap-ci-artifacts-master-gpu-operator-e2e/1418320896631771136

**Comment by kpouget** [comment]
@omertuc I think we can merge this one, I'll test & fix the flakes on https://github.com/openshift-psap/ci-artifacts/pull/224 (NFD & GPU-Operator packagemanifest not being available)

**Pull request merged**